### PR TITLE
Add support for Tracking Domains api

### PR DESCRIPTION
**Pull request by optii**
Hi,

This Pull Requests add support for the missing [tracking domain API](https://developers.sparkpost.com/api/tracking-domains/#header-tracking-domain-object)

Thanks,

### CODE REVIEW

File: lib/trackingDomains.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      };
+
+      return client.post(options, callback);
+    },
+
+    /**
+     * Update an existing tracking domain
+     *
+     * @param {string} domain - The domain to update
+     * @param {Object} updateOpts - Hash of the sending domain attributes to update
```
Comment (jgzamora) [inline lib/trackingDomains.js]: remove "sending"

File: test/spec/trackingDomain.spec.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    });
+
+    it('should throw an error if create options are missing', function() {
+      return expect(trackingDomains.create()).to.be.rejectedWith('create options are required');
+    });
+  });
+
+  describe('update', function() {
+    it('should call client put method with the appropriate uri and payload', function() {
+      var sendingDomain = {
```
Comment (jgzamora) [inline test/spec/trackingDomain.spec.js]: this var name is a bit confusing, trackingDomain or just domain could work

**Comment by optii** [comment]
Made the changes as suggested!

**Comment by optii** [comment]
Following up on this, is there anything stopping this from being merged ?

**Comment by optii** [comment]
Hi, just checking in on this, is there anything else needed for this to get merged ? Happy to do the necessary to get this through.

Thanks,

**Pull request closed without merging**